nixops/backends/__init__.py

```diff
@@ -453,15 +454,62 @@ def get_keys(self):
         return self.keys
 
     def get_ssh_name(self) -> str:
+        """
+        In ssh terminology, this is the "Host", which part of the "destination"
+        but not necessarily the same as the "Hostname".
+        The ssh config file can set Hostname for specific Hosts, effectively
+        rewriting the destination into the final hostname or ip.
+        """
         assert False
 
-    def get_ssh_flags(self, scp: bool = False) -> List[str]:
-        if scp:
-            return ["-P", str(self.ssh_port)] if self.ssh_port is not None else []
+    def get_ssh_host_keys(self) -> Optional[str]:
+        """
+        Return the public host key in known_hosts format or None if not known.
+        """
+        return None
+
+    def _get_ssh_ambient_options(self) -> Dict[str, str]:
+        proc = subprocess.Popen(
+            ["ssh", "-G", self.get_ssh_name()], stdout=subprocess.PIPE
+        )
+        opts: Dict[str, str] = {}
+        if proc.stdout is None:  # mostly for mypy; Popen won't do this to us.
+            return opts
```

Comment:
This would be a bug:

```suggestion
        assert proc.stdout is not None  # mostly for mypy; Popen won't do this to us.
```